### PR TITLE
Add presets for github credentials to project infra 

### PR DESCRIPTION
**Pull request by brianmcarey**
Add presets for github credentials to project infra presubmit

Signed-off-by: Brian Carey <bcarey@redhat.com>

### CODE REVIEW

File: github/ci/prow-deploy/files/jobs/kubevirt/project-infra/project-infra-periodics.yaml [hunk 1]
```diff
@@ -35,17 +37,12 @@ periodics:
       - --template
       - --ceiling=1
       - --confirm
-      volumeMounts:
-      - name: token
-        mountPath: /etc/github
-    volumes:
-    - name: token
-      secret:
-        secretName:  commenter-oauth-token
```
Comment (dhiller) [inline github/ci/prow-deploy/files/jobs/kubevirt/project-infra/project-infra-periodics.yaml:44]: I think that this is a different token than what we are using everywhere else. Can you check please? If that is the case, you might create a separate preset (i.e. `commenter-github-token`)

File: github/ci/prow-deploy/files/jobs/kubevirt/project-infra/project-infra-presubmits.yaml [hunk 2]
```diff
@@ -375,9 +372,6 @@ presubmits:
       - name: pgp-bot-key
         secret:
           secretName: pgp-bot-key
-      - name: unprivileged-token
-        secret:
-          secretName: unprivileged-oauth-token
```
Comment (dhiller) [inline github/ci/prow-deploy/files/jobs/kubevirt/project-infra/project-infra-presubmits.yaml:380]: I think we should not touch the job config for this job, it actually uses the unprivileged token to not cause any interference when testing the prow instance within the kind cluster IIRC.
Especially not fetching GitHub events for the real prow instance (which is what happened when we first tried this accidentally with the same token as the production prow uses) and also not messing with PR status then.

File: github/ci/prow-deploy/kustom/base/configs/current/config/config.yaml [hunk 3]
```diff
@@ -538,6 +538,20 @@ presets:
...
+    preset-github-credentials: "true"
+  env:
+  - name: GIT_ASKPASS
+    value: ../project-infra/hack/git-askpass.sh
+  - name: GITHUB_TOKEN
+    value: "/etc/github/oauth"
+  volumes:
+  - name: github-token
+    secret:
+      secretName: oauth-token
```
Comment (dhiller) [inline github/ci/prow-deploy/kustom/base/configs/current/config/config.yaml:551]: Maybe we can set user readonly, but not sure whether that would work for every job?

```suggestion
      defaultMode: 0400
      secretName: oauth-token
```

**Review by dhiller — COMMENTED** [review]
Hey, looks very good already! Some comments inline.

File: github/ci/prow-deploy/files/jobs/kubevirt/project-infra/project-infra-periodics.yaml (hunk 1 above)
Comment (brianmcarey) [reply github/ci/prow-deploy/files/jobs/kubevirt/project-infra/project-infra-periodics.yaml:44]: yeah I will double check this.

Comment (brianmcarey) [reply github/ci/prow-deploy/files/jobs/kubevirt/project-infra/project-infra-periodics.yaml]: The tokens were different so I added commenter-oauth-token and unprivileged token back in latest update.

File: github/ci/prow-deploy/kustom/base/configs/current/config/config.yaml (hunk 3 above)
Comment (brianmcarey) [reply github/ci/prow-deploy/kustom/base/configs/current/config/config.yaml:551]: Tested a couple of jobs and they succeeded with the above suggested permissions.

**Review by dhiller — APPROVED** [review]
/approve

**Pull request merged**